### PR TITLE
Update spec documentation

### PR DESCRIPTION
**Pull request by dannyd4315**
This PR just updates the documentation for specs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -244,7 +244,34 @@ def test_publish_received_payment
   )
 end
 ```
+### Stubbing Kafka requests
 
+If you are using the `:essential` and `:batched` priorities for your events you will need to stub requests to your `test` Kafka service as these priorities require an acknowledgement from Kafka that it has received the sent message.
```
Comment (balvig) [inline README.md]: Oh, I wonder why this is? 

In testing I assume we would be using the `TestMessageBus`, which should simply add the message (along with priority) to the `deliveries` hash?

File: README.md [hunk 2]
```diff
@@ -244,7 +244,34 @@ def test_publish_received_payment
...
+
+```ruby
+RSpec.configure do |config|
+  config.before(:each) do
+    require "avro_turf/test/fake_confluent_schema_registry_server"
+    Streamy.configuration.avro_schema_registry_url = "http://registry.example.com"
+    Streamy.configuration.avro_schemas_path = "app/events/schemas"
+    stub_request(:any, /^#{Streamy.configuration.avro_schema_registry_url}/).to_rack(FakeConfluentSchemaRegistryServer)
+  end
+end
```
Comment (balvig) [inline README.md]: 👌 (maybe we can consider adding a helper for this as well)

File: README.md (hunk 1 above)
Comment (dannyd4315) [reply README.md]: Unfortunately you need to have a broker set up in config, we are not able to remove the line you described. We could achieve the same result by removing `super` from the TestDispatcher, however, this then would not be testing the avro encoding and ensuring schemas are correct.

File: README.md (hunk 2 above)
Comment (dannyd4315) [reply README.md]: 👍 I can move this to an Avro helper.

File: README.md (hunk 1 above)
Comment (dannyd4315) [reply README.md]: Alternately we can set up a fake broker as ruby-kafka does, but to me that seems over-kill as discussed here - https://github.com/cookpad/streamy/pull/87#issuecomment-513170144

What do you think?

Comment (balvig) [reply README.md]: > Unfortunately you need to have a broker set up in config due to `ruby-kafka`, we are not able to remove the line you described.

Oh weird, I swear I had that working somehow in the tests, let me take another look.

File: README.md (hunk 2 above)
Comment (balvig) [reply README.md]: Maybe we can wait until we need to send events from another app (and add tests for them).
Then probably what works well will reveal itself?

File: README.md (hunk 1 above)
Comment (balvig) [reply README.md]: @dannyd4315 I must have been dreaming that... 😅 🙇 

What do you think about something like https://github.com/cookpad/streamy/pull/87/commits/302776474d12c988f60790d5eecf3a89f420f1d2?

it goes with the "don't call `super` approach" you mentioned, but also gives us a way to inspect that messages actually get encoded correctly.

That way we are one step away from actually sending out real messages, and still have the capability to inspect both events and their corresponding encoded messages separately.

Does that make sense...? 🤔

Comment (dannyd4315) [reply README.md]: Yes you are 100% correct, we don't actually need to send the real message out in tests, all we care about is whether it is encoded correctly and the schema is correct. We test everything else in Streamy & ruby-kafka. Just running the updated branch on `global-web` now, but looks good 👍

File: README.md (hunk 2 above)
Comment (dannyd4315) [reply README.md]: Sounds good

File: README.md (hunk 1 above)
Comment (dannyd4315) [reply README.md]: Tested on `global-web` and removed the stubbing details in the readme as not required any more.

**Comment by dannyd4315** [comment]
♻️

**Pull request merged**